Dockerfile.dev

```diff
@@ -0,0 +1,34 @@
+FROM adoptopenjdk/openjdk11-openj9:jre-11.0.8_10_openj9-0.21.0
+
+## Build image with sensible default environment values ##
+ENV gapsVersion=*

```

Comment:
I had to set the version here otherwise I get `gaps_dev | Error: Unable to access jarfile ./GapsWeb-.jar`

Reply:
This should be resolved with latest commit. Bad clean-up job on my part.

Reply:
I still get the same error when I run. I'm on docker-compose version 1.27.4, build 40524192

Reply:
We are on the same version of docker-compose. I'm on Docker v19.03.14, build 5eb3275d40. So far to verify things are working as expected on my end I have ran the following.

To verify the git commit I am using personally:
```42a8281 (HEAD -> improvement/simplify-dev-docker-flow, origin/improvement/simplify-dev-docker-flow) Comment out missed file to make compose file correct```

To verify without the usage of docker-compose I've done the following:
`$ sudo docker system prune -a`
`$ sudo docker build -f Dockerfile.dev -t gaps_dev .`
`$ sudo docker run -v /....../gaps/GapsWeb/target:/usr/app  gaps_dev:latest`
This works as expected. Note: you must provide the full path of the directory you want to mount here from the root.

To verify with the usage of docker-compose I've done the following:
`$ sudo docker system prune -a`
`$ sudo docker-compose -f docker-compose-dev.yaml up -d`
This also works as expected. 

Something to keep in mind is it's mounting the resulting target directory located at `gaps/GapsWeb/target` into the container. If the target directory does not exist, or if a Jar is not within that directory following a naming scheme of `GapsWeb-(some text).jar` you will get a similar error to the above.

With all the above being said, it seems like the issue you are running into is an envvar issue since it's looking for a file without any kind of version identifier. Either an `*` must be used to match a jar with any version identifier, or you must specify the exact version identifier of the jar in your target dir. To verify it's purely an envvar issue I suggest changing line 34 in Dockerfile.dev from `./GapsWeb-${gapsVersion}.jar` to `./GapsWeb-0.8.6.jar` or whatever your built jars name is.

Let me know - I'd be happy to use a more realtime communication tool to assist if you want.

Reply:
That worked. I think the issue was the prune to force the rebuild. I'm not as versed in compose as regular docker. Maybe we can just update the compose instructions with what you wrote above. Thanks for all the hard work on this.

Reply:
Additionally you can force a rebuild of the image with docker-compose by running `docker-compose build -f docker-compose-dev.yaml`. I think that was likely what had happened. Due to the image being built from a bad Dockerfile initially, and then since it already existed when you ran the `up` command after pull the latest changes it was still using an outdated image. I'll update the instructions to include a command to force a rebuild of the image if it already exists and that should cover all bases I believe. 

I'll leave resolution of this conversation to you once I've update the instructions in the compose file.

Reply:
Did you push the changes? I didn't see an update in the files.

Reply:
Apparently not. Pushed